### DISCUSSION TITLE
Bias word selection by user ngram error

### DISCUSSION THREAD

**Discussion by nicofirst1** [monkeytypegame/monkeytype#1138] [Ideas]
The hypothesis behind this idea is that certain combination of letters need different finger movements, some of these movements may bring the finger position outside a range in which the user is quickly able to recover. 

This causes the user to input an incorrect key.

To alleviate this you could extract an Ngram with a fixed N from the incorrect word and log it. 
After some time you'll have a mapping between incorrect ngrams and number of times the user had problems with it and you can bias the random word generator to contain more of such ngrams.

Would this be possible/reasonable?

**Comment by unrealapex** [comment]
> The hypothesis behind this idea is that certain combination of letters need different finger movements, some of these movements may bring the finger position outside a range in which the user is quickly able to recover.

Wouldn't this vary because the place of which keys are located are not constant on all keyboard layouts ?

I believe it would be possible, but it would require someone to do so for all the keyboard layouts Monkeytype supports.

**Reply by nicofirst1** [reply]
Since the algorithm would adapt to the user, I think there is no need to hand-coded it. 
And if the user switches keyboard then the ngrams will change so the feedback changes accordingly.

**Comment by Miodec** [comment]
Basic version of this should be in the works: #44
